app/actions/websocket/index.ts

```diff
@@ -168,7 +170,9 @@ export function doReconnect(now: number) {
                         if (!stillMemberOfCurrentChannel || !channelStillExists || (!viewArchivedChannels && channelStillExists.delete_at !== 0)) {
                             EventEmitter.emit(General.SWITCH_TO_DEFAULT_CHANNEL, currentTeamId);
                         } else {
-                            dispatch(getPosts(currentChannelId));
+                            const postIds = getPostIdsInChannel(state, currentChannelId);
+                            const since = getChannelSinceValue(state, currentChannelId, postIds);
+                            dispatch(getPostsSince(currentChannelId, since));
```

Comment:
I would use `loadPostsIfNecessaryWithRetry` instead as is calling getPost or getPostSince as needed, thoughts?

cc: @migbot

Reply:
I initially used that, which led me to believe that there was something different under the hood, but it also felt as too much stuff to do on a reconnect. Since the network might be unstable and be on and off, having retries on top of that seems like to much going on. if it fails is because the network went off again and it'll reconnect and try again so I think there is no need to retry, but I'm also 1/5 towards this.